### PR TITLE
#162701971 Build REST End point for User Login

### PR DESCRIPTION
**Pull request by Daymorelah**
#### What does this PR do?
Build REST end-point to handle user login feature
#### Description of Task to be completed?
* Build login controller to handle a **POST** request to `/auth/login` URL
* Write unit tests for the login controller
* Write integration test for the login controller
* Test route with postman to ensure controller is working as expected
#### How should this be manually tested?
* Fork the repo and `cd` into the project folder
* Run `npm instal` to install all required packages
* Run 'npm test` to run tests on the app.
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/162701971

### CODE REVIEW

File: src/seeders/20181217155521-testUsers.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    firstName: 'John',
+    lastName: 'Doe',
+    email: 'yomizy@wizzy.com',
+    roleId: 1,
+    password: 'myPassword',
+    username: 'icecream',
+    createdAt: new Date(Date.now()),
+    updatedAt: new Date(Date.now())
+  }], {}),
+  /*
```
Comment (kingisaac95) [inline src/seeders/20181217155521-testUsers.js]: Change filename to match naming convention; Remove comments

File: src/controller/userController.js [hunk 2]
```diff
@@ -379,18 +378,66 @@ class UserController {
...
+          };
+          return res.status(200).json({
+            success: true,
+            message: 'Login successful',
+            userDetails,
+          });
+        }
+        return helperMethods.serverError(res);
+      }
+      return res.status(400).send({
```
Comment (ah-med) [inline src/controller/userController.js:440]: @Daymorelah you did a good job here. However, I feel you should also check if the user is verified before creating the token.

File: src/controller/userController.js [hunk 3]
```diff
@@ -379,18 +378,66 @@ class UserController {
...
+   * Route: POST: auth/login
+   * @param {object} req - Request object
+   * @param {object} res - Response object
+   * @return {res} res - Response object
+   * @memberof UserController
+   */
+  static async userLogin(req, res) {
+    const { email, password } = req.body;
+    try {
+      const userFound = await Users.findOne({ where: { email, } });
```
Comment (ah-med) [inline src/controller/userController.js:405]:
```suggestion
      const userFound = await Users.findOne({ where: { email, isVerified: true} });
```

File: src/controller/userController.js [hunk 4]
```diff
@@ -106,7 +106,7 @@ class UserController {
   static async completeRegistration(req, res) {
     const verifiedToken = await Authentication.verifyToken(req.query.token);
     if (!verifiedToken) {
-      return res.status(400).send({
+      return res.status(400).json({
```
Comment (ah-med) [inline src/controller/userController.js]:
```suggestion
      return res.status(403).json({
```

File: src/seeders/20181217155521-testUsers.js (hunk 1 above)
Comment (Daymorelah) [reply src/seeders/20181217155521-testUsers.js]: The filename has been changed and all comments removed.

File: src/controller/userController.js (hunk 3 above)
Comment (Daymorelah) [reply src/controller/userController.js:405]: Thank you for pointing this out. I will add it.

Comment (Daymorelah) [reply src/controller/userController.js:405]: Checking if the user is verified has been added. Thank you once again.

File: src/controller/userController.js (hunk 2 above)
Comment (Daymorelah) [reply src/controller/userController.js:440]: Thank you for pointing this out.

**Pull request merged**